lib/mayaHydra/hydraExtensions/CMakeLists.txt

```diff
@@ -94,14 +116,43 @@ endif()
     # -----------------------------------------------------------------------------
     # link libraries
     # -----------------------------------------------------------------------------
-if(BUILD_MAYAUSD_LIBRARY)
+if(DEFINED MAYAUSD_LOCATION)
+    find_library(MAYAUSD_API_LIBRARY NAMES mayaUsdAPI PATHS ${MAYAUSD_LOCATION}/lib NO_DEFAULT_PATH)
+    if(NOT MAYAUSD_API_LIBRARY)
+        message(WARNING "Could not find mayaUsdAPI library.")
+    endif()
```

Comment:
This should be in the new FindMayaUsd.cmake

Reply:
Can use generator expression again to add the library (if found). No need for the `if(DEFINED MAYAUSD_LOCATION)`

            $<$<BOOL:${MAYAUSD_FOUND}>:${MAYAUSD_API_LIBRARY}>

Reply:
I'm pretty sure the `if(DEFINED MAYAUSD_VERSION)` can be simplified down to a single generator expression line as well. This might/should work:
`$<$<BOOL:${MAYAUSD_FOUND}>:MAYAUSD_VERSION=${MAYAUSD_VERSION}>`